lib/generator.js

```diff
@@ -35,14 +35,14 @@ module.exports = function(locals) {
   if (url[url.length - 1] !== '/') url += '/';
 
   let icon = '';
-  if (feedConfig.icon) icon = url + encodeURI(feedConfig.icon);
+  if (feedConfig.icon) icon = encodeURI(full_url_for.call(this, feedConfig.icon));
```

Comment:
`full_url_for()` [already encodes](https://github.com/hexojs/hexo-util/blob/42b7393b9c591d3874e0c687b4dba5a17512121a/lib/full_url_for.js#L24) the url, so `encodeURI()` is not needed.